install/kubernetes/helm/istio/values.yaml

```diff
@@ -418,6 +418,8 @@ grafana:
   image: grafana
   persist: false
   storageClassName: ""
+  accessModes:
```

Comment:
isn't only one access mode possible?  If so, this should be a key/value instead of a key/list.

Reply:
if so, may make sense to call this accessMode

Reply:
Well the `accessModes` property does accept multiple access modes (it is a list). I decided to make this exactly match the k8s API, even though it says in the docs that only one access mode can be used at a time. Relevant links are below. We could be opinionated and to only allow one access mode since a single one is only used, but I wanted it to support all configurations possible by the k8s spec so I opted for the list. Lmk if that makes sense.

Access Modes documentation
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

Reference Docs:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#persistentvolumeclaimspec-v1-core

Reply:
@sdake does the above make sense?

Reply:
For Kubernetes the PV list makes perfect sense in that I'd want to be able to access for Azure 1 of the 3 access modes Azure supports.  In AWS, the list that can be specified is only 1.  This results in either case (multiple options available, 1 option available) that only 1 option can be specified to the PV.  I guess a string array makes sense for the PVC (e.g. to specify two options when only one will be usable and let the backend sort it out).  I think anyone that uses this advanced functionality however will surely know what backend they are using and know how to specify the correct access mode.

Can you explain a use case that would (work) with specification of multiple access modes ?  AFAIK the only access mode that works for this particular case is readwritemany (when horizontal scaling is applied to the grafana deployment).

Cheers
-steve

Reply:
@sdake I can't think of one. Your explanation makes me lean toward being more opinionated and forcing the consumer here to choose a specific accessMode so they know what they are doing. 

The only case I think of is if someone wants to make a chart that includes istio as a subchart and it will be used on multiple cloud providers, some of which some access modes, some of which that don't. In that case too though, they could simply pass up the configuration through the helm chart, and I'm not really sold on that use case at the moment. I don't know if istio will even work properly as a subchart.

I'll update it to be more opinionated and take a single access mode for now.